Sources/SwiftKafka/KafkaTopicConfig.swift

```diff
@@ -0,0 +1,122 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+
+/// When messages get published to a non-existent topic, a new topic is created using the `KafkaTopicConfig`
+/// configuration object (only works if server has `auto.create.topics.enable` property set).
+/// `KafkaTopicConfig` is a `struct` that points to a configuration in memory.
+/// Once a property of the `KafkaTopicConfig` is changed, a duplicate in-memory config is created using the
+/// copy-on-write mechanism.
+/// For more information on how to configure Kafka topics, see
+/// [all available configurations](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md#topic-configuration-properties).
+public struct KafkaTopicConfig: Hashable, Equatable {
+    private final class _Internal: Hashable, Equatable {
+        /// Pointer to the `rd_kafka_topic_conf_t` object managed by `librdkafka`
+        private(set) var pointer: OpaquePointer
+
+        /// Initialize internal `KafkaTopicConfig` object with default configuration
+        init() {
+            self.pointer = rd_kafka_topic_conf_new()
+        }
+
+        /// Initialize internal `KafkaTopicConfig` object through a given `rd_kafka_topic_conf_t` pointer
+        init(pointer: OpaquePointer) {
+            self.pointer = pointer
+        }
+
+        deinit {
+            // rd_kafka_topic_conf_destroy(pointer)
```

Comment:
We are running into an issue here:

The `pointer` to the `rd_kafka_topic_conf_t` gets automatically freed by `rd_kafka_topic_new`. This leads us to a situation where we have a double-freeing of this pointer. However, when the `KafkaTopicConfig` is created without passing it to `rd_kafka_topic_new`, we still need to deallocate it here.

A solution could be checking if the pointer has been freed or not. Though I don't know whether this is possible.